src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -4032,9 +4057,15 @@ private SqlFedAuthToken getFedAuthToken(SqlFedAuthInfo fedAuthInfo) throws SQLSe
 
         while (true) {
             if (authenticationString.trim().equalsIgnoreCase(SqlAuthentication.ActiveDirectoryPassword.toString())) {
-                fedAuthToken = SQLServerADAL4JUtils.getSqlFedAuthToken(fedAuthInfo, user, password,
+                accessToken = SQLServerADAL4JUtils.getSqlFedAuthToken(fedAuthInfo, user, password,
```

Comment:
Can we turn the three else if statements (along with the trim + string comparison) into one switch statement?